### ISSUE TITLE
Getting synchronized updates to work

### ISSUE THREAD

**Issue by ErrorNoInternet** [tmux/tmux#4299]
### Issue description

I tried patching tmux to get synchronized updates to work (following #3325), but the screen is still flickering. The closest I got to no flickering was when I added a `screen_write_start(sctx, sctx->s)` before sending a sync start escape sequence and a `screen_write_stop(sctx)` after sending a sync stop, but this causes tmux to crash sometimes and there is corruption in the output. Is there anything else I can try? I am a bit unfamiliar with tmux internals, so any guidance is appreciated.

Patch: [tmux-diff.txt](https://github.com/user-attachments/files/18137903/tmux-diff.txt) (not sure how to send sequence to current pane only)
Test program: [t.py.txt](https://github.com/user-attachments/files/18137905/t.py.txt)

### Required information

* tmux version: `tmux 3.5a`
* Platform: Linux 6.12.4 x86_64 (`uname -sp`: `Linux unknown`)
* $TERM inside and outside of tmux: `tmux-256color`, `xterm-kitty`

Logs while running with the patch: 
[tmux-client-2333915.log](https://github.com/user-attachments/files/18137917/tmux-client-2333915.log)
[tmux-out-2333917.log](https://github.com/user-attachments/files/18137918/tmux-out-2333917.log)
[tmux-server-2333917.log](https://github.com/user-attachments/files/18137919/tmux-server-2333917.log)

**Comment by nicm** [comment]
I'm not really clear what you are trying to do? How are you testing?

I'm a bit tempted to remove support for synchronized updates because they never really worked very well and are not terribly useful...

**Comment by ErrorNoInternet** [comment]
I'm trying to make tmux pass synchronized updates from applications to the terminal emulator, but tmux seems to be doing some extra buffering that breaks this. Running `t.py` inside of tmux results in visible flickering, while it does not happen when running in a plain kitty instance.

**Comment by nicm** [comment]
Yeah it is hard to make it map directly because tmux will not necessarily draw in exactly the same way as the application does.

**Comment by marcindulak** [comment]
I'm coming here from https://github.com/anthropics/claude-code/issues/769#issuecomment-3667315590, which mentions https://github.com/contour-terminal/vt-extensions/blob/master/synchronized-output.md.

Am I in the right place to follow the progress?

**Comment by abbec** [comment]
FWIW, I have the same problem, reported (perhaps wrongfully) to Alacritty here: https://github.com/alacritty/alacritty/issues/8807

**Comment by marcindulak** [comment]
FYI: there was some potentially related code changes in https://github.com/tmux/tmux/pull/4744, commited to master in https://github.com/tmux/tmux/commit/1c7e164c22a3cb21d1e281087cd4aaa489d72b84, mentioned in https://github.com/anthropics/claude-code/issues/9935#issuecomment-3708542416.

**Comment by abbec** [comment]
> FYI: there was some potentially related code changes in https://github.com/tmux/tmux/pull/4744, commited to master in https://github.com/tmux/tmux/commit/1c7e164c22a3cb21d1e281087cd4aaa489d72b84, mentioned in https://github.com/anthropics/claude-code/issues/9935#issuecomment-3708542416.

Yeah, that seems exactly like what I am looking for! Thanks!

**Comment by nicm** [comment]
Let me know if this is still an issue with `master` after https://github.com/tmux/tmux/pull/4744.
